workflow/controller/workflowpod.go

```diff
@@ -355,6 +355,15 @@ func (woc *wfOperationCtx) createWorkflowPod(nodeName string, mainCtr apiv1.Cont
 		if err != nil {
 			return nil, errors.Wrap(err, "", "Error in Unmarshalling after merge the patch")
 		}
+
+		err = json.Unmarshal([]byte(tmpl.PodSpecPatch), &spec)
```

Comment:
where do we already do this? we must already apply the patch somewhere - how come a second apply?

Reply:
We are not merging here. Marshall the PodSpecPatch to get a container which is only patching through podSpec for adding volumeRef.